### DISCUSSION TITLE
Feature request: Gallery mode for Listeria

### DISCUSSION THREAD

**Discussion by trnstlntk** [magnusmanske/listeria_rs#52] [Ideas]
_tl;dr Would you consider creating a gallery mode for Listeria? It may become a game-changer on Wikimedia Commons._

With Listeria v1, quite a few people (including myself) have been experimenting with Wikidata-driven galleries based on Listeria, especially on Wikimedia Commons.

Some especially nice and interesting examples:
* [Self-portraits of women](https://commons.wikimedia.org/wiki/Self-portraits_of_women). This page automagically grows as more people create and edit Wikidata items of women's self portraits!
* [WikiFlix](https://commons.wikimedia.org/wiki/User:Spinster/WikiFlix/Directed_by_female_directors). Imagine a world in which Wikimedia Commons would become the Netflix of public domain and freely licensed movies!
* [1999 sculptures in the Netherlands](https://commons.wikimedia.org/wiki/1999_sculptures_in_the_Netherlands).

Until now this has functioned with the help of [custom-built row templates](https://commons.wikimedia.org/wiki/Special:PrefixIndex/Template:Wikidata_list/). I have found this solution to be quite shaky and complex. Each time I've tried to create a new 'type' of gallery, I've had to recreate a [new](https://commons.wikimedia.org/wiki/Template:Wikidata_list/Gallery_of_public_art) [type](https://commons.wikimedia.org/wiki/Template:Wikidata_list/Gallery_with_creator) [of](https://commons.wikimedia.org/wiki/Template:Wikidata_list/WikiFlix) row template (and I'm definitely not proficient in MediaWiki templates). After the release of Listeria v2, quite a few pages with these existing row templates have become buggy (see the sudden appearance of ugly {{{p18}}} code in [this page](https://commons.wikimedia.org/wiki/Self-portraits_of_women) for instance), and I expect that this may happen with any new Listeria release as well.

Wouldn't it make sense to include a native 'gallery mode' in Listeria v2? Basically a gallery is a list in a different layout...

Some things I've been thinking about, definitely not exhaustive and in some cases quite advanced:
* Should support all types of media files hosted on Wikimedia Commons: audio files, pdfs, video files, 3D files...
* Ideally built to, at some point, also support structured data on Commons. The examples I've linked above all are based on gallery views of Wikidata items, but there's definitely a major use case for galleries of Commons files. A nice example would be chronological lists of photos of Angela Merkel (or basically each famous person), e.g. grouped by decade. That depicts + date taken (structured) data lives in the MediaInfo of the files on Wikimedia Commons.
* It would be great if Listeria would give the 'gallery builder' various layout options: whether the contents of a property goes above or below the image (or perhaps even to the left or right of it?) + frivolous things like font size, bold, italic, centered or left/right aligned...
* It would be great (also as a general Listeria feature) if it would be possible to specify the 'linking behavior' for each individual property. For instance (theoretical use case [in a variant of this gallery](Rotterdam kunstwerk zwaar weer.jpg)): names of creators only locally linked, administrative location not linked, item label itself linked to Wikidata.

Thanks for considering this. Listeria is awesome!

**Comment by magnusmanske** [comment]
Will definitely put on the list, sounds fun and worthwile to do!

SDC is difficult while the SPARQL server still requires OAuth login :-(
